cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2222 @@
+      "\u001b[0;31mTypeError\u001b[0m: sequence item 1: expected str instance, dict found"
+     ]
+    }
+   ],
+   "source": [
+    "question = \"Based on the sales principles at SAP Business One, where is item information stored?\"\n",
+    "\"\".join(rag_pipeline.search(question, top_k=2, filter_query={\"topic\": {\"$eq\": \"sales\"}})[0])\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6a171971-de4f-462a-ab05-ca2b49a3b4a6",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "001bad18-bc88-46ed-8aa1-94d4ac3ee45b",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "18705b19-c9bb-49d8-a29e-e1705e7a78d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### find something in pinecone more accurate than handpicking the topic\n",
+    "\n",
+    "### use pinecone to make RAG demo more realistic "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e95be9a2-9b46-4d2b-88a3-ca3a0ad7bc51",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ee253b8-1ecf-4ea2-a423-13bbe016e11e",
+   "metadata": {},
+   "source": [
+    "### Can we find a hallucination?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 99,
+   "id": "5abcca04-a189-4d30-ab7e-63052dcb2daf",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'response': 'The given documents do not provide any information about whether the TurboCharge EV6000 supports the UL/CE/ISO charger protocols.',\n",
+       " 'trustworthiness_score': 0.6223180925974554}"
+      ]
+     },
+     "execution_count": 99,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "question = \"Does the TurboCharge EV6000 support the UL/CE/ISO charger protocols?\"\n",
+    "top_docs = \" | \".join(rag_pipeline.search(question, top_k=1, topic='product'))\n",
+    "\n",
+    "prompt = f\"Based on the following documents, answer the given question. \\n\\\n",
+    "Documents: {top_docs} \\n\\\n",
+    "Question: {question}\"\n",
+    "\n",
+    "tlm = studio.TLM(quality_preset=\"low\",)\n",
```

Comment:
should explain why you're setting low here.

should also explain that:

You can try setting the `quality_preset="best"` to get even more accurate LLM answers along with the trustworthiness scores.